server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/cancel/CancelTasksResponse.java

```diff
@@ -19,23 +19,64 @@
 
 package org.elasticsearch.action.admin.cluster.node.tasks.cancel;
 
-import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.action.TaskOperationFailure;
 import org.elasticsearch.action.admin.cluster.node.tasks.list.ListTasksResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.tasks.TaskInfo;
 
+import java.io.IOException;
 import java.util.List;
 
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
 /**
  * Returns the list of tasks that were cancelled
  */
 public class CancelTasksResponse extends ListTasksResponse {
 
+    private static final ConstructingObjectParser<CancelTasksResponse, Void> PARSER =
```

Comment:
Do you think you can make something like:
```
protected static <T> ConstructingObjectParser<T, Void> setupParser(String name, TriFunction<List<TaskInfo>, List<TaskOperationFailure>, List<ElasticsearchException>> ctor)
```
in ListTasksResponse? That way you wouldn't need to copy this stuff.

Reply:
great idea! will do

Reply:
@nik9000 I went to do this and ran into a problem: `setupParser` would maybe need to instantiate a response: `return new ListTasksResponse(tasks, tasksFailures, nodeFailures);`,  someplace, were you thinking that the calling method would supply a lambda to do this?

because otherwise, unless I'm missing something, couldn't call `new` on a generic type. Am I missing something?

Reply:
I'd supply a lambda, yeah.

Reply:
@nik9000 nice suggestion, this is implemented now